library/src/main/java/io/ona/kujaku/helpers/OfflineServiceHelper.java

```diff
@@ -0,0 +1,110 @@
+        intent.putExtra(Constants.PARCELABLE_KEY_STYLE_URL, mapboxStyleUrl);
+        intent.putExtra(Constants.PARCELABLE_KEY_MAP_UNIQUE_NAME, mapName);
+        intent.putExtra(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN, mapBoxAccessToken);
+        intent.putExtra(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND, topLeftBound);
+        intent.putExtra(Constants.PARCELABLE_KEY_TOP_RIGHT_BOUND, topRightBound);
+        intent.putExtra(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND, bottomRightBound);
+        intent.putExtra(Constants.PARCELABLE_KEY_BOTTOM_LEFT_BOUND, bottomLeftBound);
+        intent.putExtra(Constants.PARCELABLE_KEY_MIN_ZOOM, zoomRange.getMinZoom());
+        intent.putExtra(Constants.PARCELABLE_KEY_MAX_ZOOM, zoomRange.getMaxZoom());
+
+        context.startService(intent);
+    }
+
+    /**
+     * Requests the deletion of an offline map from {@Link MapboxOfflineDownloaderService}. This is
+     * an offline map already downloaded or queued to download using the {@link MapboxOfflineDownloaderService}
+     *
+     * @param context
+     * @param mapName
+     * @param mapBoxAccessToken
+     */
+    public static void deleteOfflineMap(@NonNull Context context, @NonNull String mapName, @NonNull String mapBoxAccessToken) {
+        Intent intent = new Intent(context, MapboxOfflineDownloaderService.class);
+        intent.putExtra(Constants.PARCELABLE_KEY_SERVICE_ACTION, MapboxOfflineDownloaderService.SERVICE_ACTION.DELETE_MAP);
+        intent.putExtra(Constants.PARCELABLE_KEY_MAP_UNIQUE_NAME, mapName);
+        intent.putExtra(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN, mapBoxAccessToken);
+
+        context.startService(intent);
+    }
+
+    /**
+     * Requests the {@Link MapboxOfflineDownloaderService} to stop downloading the current map referenced
+     * by the {@code mapName} passed. This will also download all the resources of the
+     * {@link com.mapbox.mapboxsdk.offline.OfflineRegion} downloaded that are not shared with other {@link com.mapbox.mapboxsdk.offline.OfflineRegion}
+     *
+     * @param context
+     * @param mapName
+     * @param mapBoxAccessToken
+     */
+    public static void stopMapDownload(@NonNull Context context, @NonNull String mapName, @NonNull String mapBoxAccessToken) {
+        Intent stopDownloadIntent = new Intent(context, MapboxOfflineDownloaderService.class);
+        stopDownloadIntent.putExtra(Constants.PARCELABLE_KEY_SERVICE_ACTION, MapboxOfflineDownloaderService.SERVICE_ACTION.STOP_CURRENT_DOWNLOAD);
+        stopDownloadIntent.putExtra(Constants.PARCELABLE_KEY_MAP_UNIQUE_NAME, mapName);
+        stopDownloadIntent.putExtra(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN, mapBoxAccessToken);
+        stopDownloadIntent.putExtra(Constants.PARCELABLE_KEY_DELETE_TASK_TYPE, MapBoxOfflineQueueTask.TASK_TYPE_DOWNLOAD);
+
+        context.startService(stopDownloadIntent);
+    }
+
+    public static class ZoomRange {
```

Comment:
Any reason why `ZoomRange` should be nested in this class, yet it's public static?

Is the idea that all instances of the `OfflineServiceHelper` will use the same zoom range?

Reply:
This looks more like a utility class. The fact that it will need to be accessed outside of an `OfflineServiceHelper` instance suggests that it can be a class on it's own and may find use in another case scenario?

Usually, inner classes are private if only needed by that class or public if interfaces.

Is it possible to make it a util class?

Reply:
```
1. It is a way of logically grouping classes that are only used in one place: If a class is useful to only one other class, then it is logical to embed it in that class and keep the two together. Nesting such "helper classes" makes their package more streamlined.

2. It increases encapsulation: Consider two top-level classes, A and B, where B needs access to members of A that would otherwise be declared private. By hiding class B within class A, A's members can be declared private and B can access them. In addition, B itself can be hidden from the outside world.

3. It can lead to more readable and maintainable code: Nesting small classes within top-level classes places the code closer to where it is used.
```

I think point 1. is the only one that could warrant some merit here, since the inner class does not really manipulate any private variables. However, it is still not convincing that a general concept like zoom range can only find use in this one class. The Android examples you gave are clearly specific to the use cases. A builder class should ideally only build one type of object and the `ShortcutIconResource` is only related to firing of intents to launch apps, each would/should clearly only ever be used in these contexts.

However, it's not such an important detail and should not drag this on.